### PR TITLE
C-ext fixes

### PR DESCRIPTION
**Pull request by eregon**

### CODE REVIEW

File: lib/cext/ruby.h [hunk 1]
```diff
@@ -101,11 +101,11 @@ VALUE rb_tr_hidden_variable_get(VALUE object, const char *name);
 
 // Handles
 
-void *rb_tr_handle_for_managed(void *managed);
-void *rb_tr_handle_for_managed_leaking(void *managed);
-VALUE rb_tr_managed_from_handle_or_null(void *handle);
-void *rb_tr_managed_from_handle(void *handle);
-void rb_tr_release_handle(void *handle);
+void* rb_tr_handle_for_managed(VALUE managed);
+void* rb_tr_handle_for_managed_leaking(VALUE managed);
```
Comment (chrisseaton) [inline lib/cext/ruby.h]: Why did you make this change? This is the opposite of what we do everywhere else - the `*` is part of the name, not the type, in C.

Comment (eregon) [inline lib/cext/ruby.h]: It is part of the type, semantically at least.
I want to make it clear these functions convert from native (void*) to managed (VALUE) or the opposite. Of course this is only for the reader as they are typedef of each other.
Having the `*` separated makes it harder to see that in my opinion.

Comment (chrisseaton) [inline lib/cext/ruby.h]: Well I just disagree with the style change, but anyway you didn't even change it for the rest of the file so it's just inconsistent now - you've got mixed `void *` and `void*`.

Comment (chrisseaton) [inline lib/cext/ruby.h]: It looks like MRI is `void *name`, so let's stick with that.

Comment (eregon) [inline lib/cext/ruby.h]: I find
```c
void *rb_tr_handle_for_managed(VALUE managed);
```
much less clear on being a conversion function. But it is inconsistent, indeed.

Comment (eregon) [inline lib/cext/ruby.h]: Fixed.

File: lib/cext/ruby.h [hunk 2]
```diff
@@ -78,7 +78,7 @@ extern "C" {
 
 // Basic types
 
-typedef void *VALUE;
+typedef void* VALUE;
 typedef long SIGNED_VALUE;
```
Comment (chrisseaton) [inline lib/cext/ruby.h]: There's still this one.

Comment (eregon) [inline lib/cext/ruby.h]: That one is horribly confusing, do you really want it?
It's like if we alias `void` and `*VALUE`.

Comment (chrisseaton) [inline lib/cext/ruby.h]: That's the way I've always seen it written. Compare `typedef void*` and `typedef void *` in MRI - the second one is the way they do it most of the time.

Comment (eregon) [inline lib/cext/ruby.h]: Done.

**Pull request merged**